### DISCUSSION TITLE
CommandInteractionOptionResolver .has methods

### DISCUSSION THREAD

**Discussion by Floffah** [discordjs/discord.js#6145] [Q&A] [answered]
so i saw that slash command options in the form of a collection were replaced by the CommandInteractionOptionResolver, this only has .get methods, will this ever get .has methods or are they redundant?

**Comment by Jiralite** [comment, marked as answer]
The many `.get()` methods on the `CommandInteractionOptionResolver` all return `null` if your specified option does not exist. Does this help you?

Before you could have done:
```JavaScript
const options = <CommandInteraction>.options;

if (options.has("option name")) {
  // There exists a specified option name!
}
```
Presently, you can do this:
```JavaScript
const options = <CommandInteraction>.options;

if (options.get("option name") !== null) {
  // There exists a specified option name!
}
```

**Reply by Floffah** [reply]
Yeah, thats what im using for now, i just thought .has might make more sense especially since the .get methods have a required parameter

**Reply by Jiralite** [reply]
That parameter is `false` by default, but if chosen to be `true`, It will run `throw new TypeError('COMMAND_INTERACTION_OPTION_NOT_FOUND', "option name");` so it'll actually throw a `TypeError` instead of returning just `null`!
